src/io/tree.cpp

```diff
@@ -310,6 +310,14 @@ std::string Tree::ToIfElse(int index, bool is_predict_leaf_index) {
   if (num_leaves_ == 1) {
     str_buf << "return 0";
   } else {
+    str_buf << "std::vector<uint32_t> cat_threshold = {";
```

Comment:
it seems const variate is enough: https://stackoverflow.com/questions/4748083/when-should-you-use-constexpr-capability-in-c11